packages/slice-machine/server/src/api/slices/delete.ts

```diff
@@ -0,0 +1,158 @@
+import { BackendEnvironment } from "../../../../lib/models/common/Environment";
+import * as Libraries from "@slicemachine/core/build/libraries";
+import {
+  CustomPaths,
+  GeneratedPaths,
+} from "@slicemachine/core/build/node-utils/paths";
+import * as IO from "../../../../lib/io";
+import { remove as removeSliceFromMockConfig } from "../../../../lib/mock/misc/fs";
+import path, { resolve } from "path";
+
+import onSaveSlice from "../common/hooks/onSaveSlice";
+import { MocksConfig } from "../../../../lib/models/paths";
+import { DeleteSliceResponse } from "../../../../lib/models/common/Slice";
+
+interface DeleteSliceBody {
+  sliceId: string;
+  libName: string;
+}
+
+export async function deleteSlice(req: {
+  body: DeleteSliceBody;
+  env: BackendEnvironment;
+}): Promise<DeleteSliceResponse> {
+  const { env } = req;
+  const { sliceId, libName } = req.body;
+
+  if (!env.manifest.libraries) {
+    const message = `When deleting slice: ${sliceId}, there were no libraries configured in your SM.json.`;
+    console.error(`[slice/delete] ${message}`);
+    return {
+      err: new Error(message),
```

Comment:
Does the `Error` wrapper add value?

Reply:
Not too sure. This is how it's done in other responses so I thought I'd keep it consistent